src/LayoutComponents/search/redux/searchEpic.js

```diff
@@ -22,9 +23,23 @@ function searchTermEpic(actions$, { getState }) {
     )
     .flatMap(() => {
       const { searchTerm } = getState().search;
-      if (searchTerm.length > 2) {
+      // if user is pressing backspace
+      // increase debounce/throttle time
+      if (searchTerm.length < previousSearchTerm.length) {
```

Comment:
The throttle and debounce functions are set up on the first run on this function (page load) and are never run again. Therefore, changing the `Xms` doesn't have any effect.

Reply:
Good catch!

Reply:
I have moved Xms outside of the scope of the epic so it should work as expected now.

Reply:
@Bouncey I added `console.log('run')` to line 19 as a test. It only logged on page load, and not after it. That would mean the function is only fired once, so `Observable.merge` is also only called once, which in turn means that while the variable may change, because the function isn't called again with the new parameter the delays are never updated. `flatMap`'s callback is the only thing that is run repeatedly.

Reply:
Ok 👍